Issue/woomob 506 lack of contrast on blaze main ctas in dark mode

JorgeMucientes opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes WOOMOB-506
Closes WOOMOB-508

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
Fixes several accessibility issues. Most of them aimed at fixing the issues on the Blaze purchase flow, but as a side effect of fixing colors in the root components, we are fixing multiple issues across the app. The following document: 1nxDAV51DK_l8opZ2ZznqaA3FpfiYPoWZPd38HA4TTys-gdoc can be used as a reference to most of the accessibility issues that were addressed in this PR. 

The PR focuses on fixing: 
- Insufficient color contrast issues (mainly on buttons and text)
- Layouts are breaking due to the big font size used.

### Testing information
The exhaustive way to check this is to use Android Studio's "UI Check Mode" tool over composable previews for all Blaze screens and check that there aren't any color contrast issues around buttons or other UI like image placeholders or errors. 

Also, ensure no UI elements become "broken" when setting a big font size. 

### The tests that have been performed
<!-- To give the reviewer an idea of what could be missed in terms of testing -->
The above ☝🏼 

- [ ] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

JorgeMucientes committed:
Ensure Edit CTA is visible when font size is big

JorgeMucientes committed:
Update blaze product placeholder to increase color contrast

JorgeMucientes committed:
Update Woo primary button background to fix contrast issues

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is assigned to the mil...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/259">22.5</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

JorgeMucientes committed:
Increase color contrast for errors

JorgeMucientes committed:
Remove hardcoded reference to purple use semantic color instead

JorgeMucientes committed:
Increase color contrast for disabled text buttons

JorgeMucientes committed:
Increase color contrast for disabled text buttons

Use surface medium instead

irfano commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/budget/BlazeCampaignBudgetScreen.kt:
```diff
@@ -331,27 +331,21 @@ private fun CampaignDurationRow(
-            Spacer(modifier = Modifier.weight(1f))
-            WCTextButton(onClick = onEditDurationTapped) {
+                    else -> "$formattedStartDate - $formattedEndDate"
+                },
```
For binary conditions, I prefer using `if`.

> Prefer using if for binary conditions instead of when - [if versus when﻿](https://kotlinlang.org/docs/coding-conventions.html#if-versus-when)

JorgeMucientes replied:
Oh I didn't see this was part of a kotlin conding convention. I actually prefer the opposite, I think `when` allows for a much more concise/readable syntax for simple conditions. 
Anyway, given it's the convention, I'll apply it from now on. Thanks for sharing.

irfano committed:
Update placeholder colors

The colors used in the Blaze campaign product placeholder are updated to use theme colors.

irfano committed:
Add Blaze product placeholder for night mode

This commit introduces a vector drawable for the Blaze campaign product placeholder specifically for night mode.

irfano commented on WooCommerce/src/main/res/drawable/blaze_campaign_product_placeholder.xml:
```diff
@@ -1,22 +1,18 @@
-            android:fillColor="#F0F0F0"
-            android:pathData="M58.574,14.724H45.86V8.606H43.355C41.653,8.606 40.936,7.745 40.936,6.147C40.936,4.549 41.653,3.689 43.355,3.689C44.207,3.689 47.721,3.661 47.721,3.661V0L42.545,0.029C38.868,0.029 36.734,2.499 36.734,6.147C36.734,9.796 38.754,12.14 42.239,12.274V14.724H29.424C29.424,14.724 14.727,21.597 14.727,22.021H73.369C73.369,21.597 58.57,14.724 58.57,14.724H58.574Z" />
-
+            android:fillColor="@color/woo_purple_10"
```
I updated the hardcoded colors in this vector to use theme colors. This makes it easier to update theme colors across the app next time. 136f22890fbbd6fb9c93fe7141018db80295ddd5

irfano commented on WooCommerce/src/main/res/drawable-night/blaze_campaign_product_placeholder.xml:
I added the night version of this vector. c9fc4d4104544be452a9179967f0dc7865560072

JorgeMucientes replied:
Hey @irfano, I just noticed that this was actually not needed. We display the product placeholder over a white background in both light and dark mode. What we are displaying is like a "preview" of how the ad will look on a website, so we don't change the background color when selecting dark mode. 
With the new dark mode placeholder, there's a bit less contrast, but I checked, and we are still within accessible contrast ratios. So it's not a big deal, we can keep it as is. 

![Screenshot 2025-05-30 at 09 09 02](https://github.com/user-attachments/assets/79a612f2-a592-48e6-98ae-b552cb41ffc0)

irfano replied:
Good catch! I missed that. 😅

irfano reviewed (approved):
Good job! Thanks for handling this.
I added an np comment and committed some minor changes. I’m approving but not merging, in case you want to respond to my comment about the if condition.
I’m approving, and this PR is good to merge any time.

JorgeMucientes committed:
Replace when for if else in binary condition to follow Koltin convention
